sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1759,6 +1759,162 @@ def _df_getitem_unicode_idx_impl(self, idx):
+            raise IndexingError('Index is out of bounds for axis')
+    """
+    func_lines = ['def _df_getitem_list_iloc_impl(self, idx):',
+                  '  check_idx = False',
+                  '  for i in idx:',
+                  '    if -1 < i < len(self._dataframe.index):',
+                  '      check_idx = True',
+                  '  if check_idx == True:']
+    results = []
+    index = '[self._dataframe._index[i] for i in idx]'
+    if isinstance(self.index, types.NoneType):
+        index = 'idx'
+    name = df_name_codegen_iloc(self)
+    for i, c in enumerate(self.columns):
+        result_c = f"result_{i}"
+        func_lines += [f"    data_{i} = pandas.Series(self._dataframe._data[{i}])",
+                       f"    {result_c} = data_{i}.iloc[numpy.array(idx)]"]
+        results.append((c, result_c))
+    data = ', '.join(f'"{col}": {data}' for col, data in results)
+    func_lines += [f"    return pandas.DataFrame(data={{{data}}}, index={index})",
+                   f"  raise IndexingError('Index is out of bounds for axis')"]
+
+    func_text = '\n'.join(func_lines)
+    global_vars = {'pandas': pandas, 'numpy': numpy, 'IndexingError': IndexingError}
+
+    return func_text, global_vars
+
+
+def df_getitem_list_bool_iloc_codegen(self, idx):
+    """
+    Example of generated implementation:
+        def _df_getitem_list_bool_iloc_impl(self, idx):
+            if len(self._dataframe.index) == len(idx):
+                data_0 = self._dataframe._data[0]
+                result_0 = pandas.Series(data_0[numpy.array(idx)])
+                data_1 = self._dataframe._data[1]
+                result_1 = pandas.Series(data_1[numpy.array(idx)])
+                return pandas.DataFrame(data={"A": result_0, "B": result_1},
+                    index=self._dataframe.index[numpy.array(idx)])
+            raise IndexingError('Item wrong length')
+    """
+    func_lines = ['def _df_getitem_list_bool_iloc_impl(self, idx):']
+    results = []
+    index = 'self._dataframe.index[numpy.array(idx)]'
+    func_lines += ['  if len(self._dataframe.index) == len(idx):']
+    name = df_name_codegen_iloc(self)
+    for i, c in enumerate(self.columns):
+        result_c = f"result_{i}"
+        func_lines += [f"    data_{i} = self._dataframe._data[{i}]",
+                       f"    {result_c} = pandas.Series(data_{i}[numpy.array(idx)])"]
+        results.append((c, result_c))
+    data = ', '.join(f'"{col}": {data}' for col, data in results)
+    func_lines += [f"    return pandas.DataFrame(data={{{data}}}, index={index})",
+                   f"  raise IndexingError('Item wrong length')"]
+
+    func_text = '\n'.join(func_lines)
+    global_vars = {'pandas': pandas, 'numpy': numpy, 'IndexingError': IndexingError}
+
+    return func_text, global_vars
+
+
+def df_name_codegen_iloc(self):
+    if isinstance(self.index, types.NoneType):
+        return 'idx'
+
+    return 'self._dataframe._index[idx]'
```

Comment:
Looks like the function should be used only one. So maybe to not define this one use use the code directly?

Reply:
The function is used 4 times in all implementations.